examples/impute/plot_multiple_imputation.py

```diff
@@ -0,0 +1,457 @@
+# datasets (Qbar). The variance of these estimates is a combination of the
+# variance of each of the m estimates (Ubar) and the variance between the m
+# estimates (B).
+#
+# Make a function that calculates Qbar from m estimates
+def calculate_Qbar(m_estimates):
+    m = len(m_estimates)
+    Qbar = 1/m * np.sum(m_estimates, axis=0)
+    return Qbar
+
+
+# Make a function that calculates T from m estimates and their variances
+def calculate_T(m_estimates, m_variances, Qbar):
+    m = len(m_estimates)
+    Ubar = 1/m * np.sum(m_variances, axis=0)
+    B = 1/(m - 1) * np.sum((Qbar - m_estimates) ** 2, axis=0)
+    T = Ubar + B + (B/m)
+
+    return T
+
+
+###############################################################################
+
+# EXAMPLE 1. COMPARE STATISTICAL ESTIMATES AND THEIR VARIANCE USING MULTIPLE
+# IMPUTATION IN A LINEAR REGRESSION MODEL.
+
+###############################################################################
+
+
+def get_results_full_dataset(X, y):
+    # Perform linear regression on full data as a way of comparison
+    estimator = RidgeCV()
+    estimator.fit(X, y)
+    y_predict = estimator.predict(X)
+
+    # Save the beta estimates, the variance of these estimates and 1.96 *
+    # standard error of the estimates. The latter is useful to know the 95%
+    # confidence interval.
+    full_coefs = estimator.coef_
+    full_vars = calculate_variance_of_beta_estimates(y, y_predict, X)
+    full_errorbar = 1.96 * np.sqrt(full_vars)
+
+    return full_coefs, full_vars, full_errorbar
+
+
+def get_results_chained_imputation(X_incomplete, y):
+    # Impute incomplete data with IterativeImputer using single imputation
+    # We perform N_ITER imputations and only use the last imputation.
+    imputer = IterativeImputer(n_iter=N_ITER,
+                               sample_posterior=True,
+                               random_state=0)
+    imputer.fit(X_incomplete)
+    X_imputed = imputer.transform(X_incomplete)
+
+    # Perform linear regression on chained single imputed data
+    # Estimate beta estimates and their variances
+    estimator = RidgeCV()
+    estimator.fit(X_imputed, y)
+    y_predict = estimator.predict(X_imputed)
+
+    # Save the beta estimates, the variance of these estimates and 1.96 *
+    # standard error of the estimates
+    chained_coefs = estimator.coef_
+    chained_vars = calculate_variance_of_beta_estimates(
+            y, y_predict, X_imputed)
+    chained_errorbar = 1.96 * np.sqrt(chained_vars)
+
+    return chained_coefs, chained_vars, chained_errorbar
+
+
+def get_results_mice_imputation(X_incomplete, y):
+    # Impute incomplete data using the IterativeImputer to perform multiple
+    # imputation. We perform N_ITER imputations and only use the last
+    # imputation, and loop this procedure m times.
+    m = 5
+    multiple_imputations = []
+    for i in range(m):
+        imputer = IterativeImputer(n_iter=N_ITER,
```

Comment:
Just call `get_results_chained_imputation(X_incomplete, y, random_state=i)` then do the pooling